maven/xml.bzl

```diff
@@ -112,31 +115,44 @@ def _find_element_start(xml, start_from):
                 if xml[cursor] == "-" and xml[cursor+1] == "-":
                     # maybe comment end?
                     if xml[cursor+2] == ">":
-                        cursor +=3
-                        cursor = xml.find("<", cursor)
+                        # found comment end
+                        cursor = xml.find("<", cursor + 3)
                         if cursor == -1:
                             return -1 # no more tags in this text.
-                        break # Found comment end, reset start and cursor and break
+                        break
                     else:
                         fail("XML comments may not contain the string \"--\". %s" % xml)
                 else:
                     cursor += 1
+        elif len(xml) - cursor > 9 and xml[cursor:cursor+9] == "<![CDATA[":
```

Comment:
Nit: it'd be nice if `"<![CDATA["` were a constant, and the magic 9's were replaced by `thatConstant.length`